src/style.js

```diff
@@ -1,20 +1,29 @@
 import { Component } from 'react'
-import StyleSheetRegistry from './stylesheet-registry'
-
-const styleSheetRegistry = new StyleSheetRegistry()
+import {
+  globalStyleSheetRegistry,
+  StyleSheetRegistryContext
+} from './stylesheet-registry'
 
 export default class JSXStyle extends Component {
   constructor(props) {
     super(props)
     this.prevProps = {}
   }
 
+  get styleSheetRegistry() {
+    return this.context
+  }
+
   static dynamic(info) {
     return info
       .map(tagInfo => {
         const baseId = tagInfo[0]
         const props = tagInfo[1]
-        return styleSheetRegistry.computeId(baseId, props)
+        // It's fine te reference globalStyleSheetRegistry here since all
+        // `computeId` does is calculate the classname (which is a pure function
+        // of baseId and props). It's only an instance method so that it can
+        // memoize the results.
+        return globalStyleSheetRegistry.computeId(baseId, props)
```

Comment:
Correct! Can you instead move methods outside of the StyleSheetRegistry class? This way we don't need to use the `globalStyleSheetRegistry` at all.

eg:

```js
export const computeId = (function createComputeId() {
  const cache = {}
  return function(baseId, props) {
    if (!props) {
      return `jsx-${baseId}`
    }

    const propsToString = String(props)
    const key = baseId + propsToString
    // return `jsx-${hashString(`${baseId}-${propsToString}`)}`
    if (!cache[key]) {
      cache[key] = `jsx-${hashString(`${baseId}-${propsToString}`)}`
    }

    return cache[key]
  }
}());
```